src/views/forms/TradeForm/TradeFormInputs.tsx

```diff
@@ -59,7 +77,9 @@ export const TradeFormInputs = () => {
         limitPriceInput: MustBigNumber(midMarketPrice).toFixed(tickSizeDecimals ?? USD_DECIMALS),
       })
     );
-  }, [dispatch, limitPriceInput, midMarketPrice, needsLimitPrice, tickSizeDecimals]);
+    // only set on market change or trade type change
```

Comment:
this is a weird way to do it

i think this useEffect is very unclear that it's actually driven by change in market and not changes in mid market price 

i think the ideal behavior would be:
- keep track of whether the user has manually written an input or not
- if the user hasnt interacted, keep updating the mid market price, otherwise, dont
- clear tracking of (1) when the market has changed

Reply:
I'd argue it is (or with minor changes could be) clear it's being driven by market and not price but your suggestion is better than mine so I did that.